src/formatServices.js

```diff
@@ -0,0 +1,81 @@
+import engine from "php-parser";
+import { readFile } from "fs/promises";
+
+export async function formatServices(rawServices, version) {
+    // Prepare to receive formatted hooks.
+    const services = [];
+
+    const parser = new engine({
+      parser: {
+        locations: true,
+        extractDoc: true,
+      },
+      ast: {
+        withPositions: true,
+        withSource: true
+      }
+    });
+
+    // Iterate over each service.
+    for (const service of rawServices) {
+
+      const [name, value] = service;
+
+      let { class: classNamespace } = value;
+
+      if (!classNamespace && value.parent) {
+        classNamespace = rawServices[`${value.parent}`]?.class;
+      }
+
+      if (!classNamespace && value.alias) {
+        classNamespace = rawServices[`${value.alias}`]?.class;
+      }
+
+      if (classNamespace && value.public !== false) {
```

Comment:
This if condition covers the service.push() as well, resulting in the missing of other service names that does not include value.public, is there a reason for this?

Missing service name examples:

- logger.channel.image
- config.storage.staging
etc....